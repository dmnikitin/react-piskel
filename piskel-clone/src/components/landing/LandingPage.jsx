import React from 'react';
import { Link } from 'react-router-dom';
import NavBar from '../navbar/Navbar';

export default function LandingPage({ isLoggedIn }) {
  return (
    <div>

      <Link to="/piskel">go to piskel</Link>
      <h1>
        Landing page
        {JSON.stringify(isLoggedIn)}
      </h1>
    </div>
  );
}
