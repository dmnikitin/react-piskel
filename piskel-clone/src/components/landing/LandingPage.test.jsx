import React from 'react'
import { shallow, mount } from 'enzyme';
import LandingPage from './LandingPage';

const wrapperShallow = shallow(<LandingPage />);
const wrapperMount = mount(<LandingPage />);

describe('landing page', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it("displays initial div", () => {
    expect(wrapperMount.find("div")).toHaveLength(3);
  });
});
