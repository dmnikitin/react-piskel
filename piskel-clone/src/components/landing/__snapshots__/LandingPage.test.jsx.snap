// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page renders correctly 1`] = `
<div
  className="landing"
>
  <div
    className="landing-overview"
  >
    <h4>
      This app is a variation of the famous
      <a
        href="http://piskelapp.com"
      >
        PISKEL
      </a>
    </h4>
    <h5>
      With this app you can:
    </h5>
    <ul>
      <li>
         draw sprites using browser Canvas-API 
      </li>
      <li>
         add, delete, duplicate as many sprites as you like 
      </li>
      <li>
         use powerful drawing tools like stroke, bucket, eraser etc
      </li>
      <li>
         set preferred pen size, canvas size, drawing color 
      </li>
      <li>
         set required framerate and watch animation preview (even in full page mode) 
      </li>
      <li>
         navigate with control keyboard buttons with option to change them 
      </li>
      <li>
         export result to filesystem as GIF 
      </li>
    </ul>
  </div>
  <div
    className="landing-screen"
  >
    <img
      alt="landing"
      height="300"
      src="src/assets/landing.jpg"
      width="500"
    />
  </div>
  <section
    className="landing-gifs"
  >
    <img
      alt="gif1"
      height="200"
      src="src/assets/1.gif"
      width="250"
    />
    <img
      alt="gif2"
      height="200"
      src="src/assets/2.gif"
      width="250"
    />
    <img
      alt="gif3"
      height="200"
      src="src/assets/3.gif"
      width="200"
    />
    <img
      alt="gif4"
      height="200"
      src="src/assets/4.gif"
      width="200"
    />
  </section>
  <footer
    className="landing-footer"
  >
     
    <h5>
      2020, Dmitrij Nikitin.
      <a
        href="https://dmnikitin.github.io/portfolio/"
      >
        portfolio
      </a>
    </h5>
  </footer>
</div>
`;
